### PR TITLE
Change #9188: Netsave now behaves like autosave

### PR DESCRIPTION
**Pull request by Taschi120**

### CODE REVIEW

File: src/saveload/saveload.cpp [hunk 1]
```diff
@@ -2982,6 +2982,43 @@ SaveOrLoadResult SaveOrLoad(const std::string &filename, SaveLoadOperation fop,
...
+ */
+void DoAutoOrNetsave(int* counter, bool netsave) {
+	char buf[MAX_PATH];
+
+	if (_settings_client.gui.keep_all_autosave) {
+		GenerateDefaultSaveName(buf, lastof(buf));
+		if (!netsave) {
+			strecat(buf, ".sav", lastof(buf));
+		}
+		else {
```
Comment (glx22) [inline src/saveload/saveload.cpp]: [Coding style](https://wiki.openttd.org/en/Development/Coding%20style)
```suggestion
		} else {
```

File: src/saveload/saveload.cpp [hunk 2]
```diff
@@ -2982,6 +2982,43 @@ SaveOrLoadResult SaveOrLoad(const std::string &filename, SaveLoadOperation fop,
...
+			strecat(buf, "-netsave.sav", lastof(buf));
+		}
+	}
+	else {
+
+		/* generate a savegame name and number according to _settings_client.gui.max_num_autosaves */
+		if (!netsave) {
+			seprintf(buf, lastof(buf), "autosave%d.sav", *counter);
+		}
+		else {
```
Comment (glx22) [inline src/saveload/saveload.cpp]: Same here

File: src/saveload/saveload.cpp [hunk 3]
```diff
@@ -2982,6 +2982,43 @@ SaveOrLoadResult SaveOrLoad(const std::string &filename, SaveLoadOperation fop,
...
+	if (_settings_client.gui.keep_all_autosave) {
+		GenerateDefaultSaveName(buf, lastof(buf));
+		if (!netsave) {
+			strecat(buf, ".sav", lastof(buf));
+		}
+		else {
+			strecat(buf, "-netsave.sav", lastof(buf));
+		}
+	}
+	else {
```
Comment (glx22) [inline src/saveload/saveload.cpp]: And here

File: src/saveload/saveload.cpp [hunk 4]
```diff
@@ -2982,6 +2982,43 @@ SaveOrLoadResult SaveOrLoad(const std::string &filename, SaveLoadOperation fop,
 	}
 }
 
+/**
+ * Create an autosave or netsave.
+ * @param counter A pointer to the counter variable to be used for rotating the file name
+ * @param netsave indicates if this is a regular autosave or a netsave
+ */
+void DoAutoOrNetsave(int* counter, bool netsave) {
```
Comment (glx22) [inline src/saveload/saveload.cpp]: Better use a reference than a pointer.

**Comment by Taschi120** [comment]
Note that this PR and PR #9397 have conflicts.

I'll fix the review comments during the weekend. Thanks for the feedback.

File: src/saveload/saveload.cpp [hunk 5]
```diff
@@ -2982,6 +2982,39 @@ SaveOrLoadResult SaveOrLoad(const std::string &filename, SaveLoadOperation fop,
 	}
 }
 
+/**
+ * Create an autosave or netsave.
+ * @param counter A reference to the counter variable to be used for rotating the file name
+ * @param netsave indicates if this is a regular autosave or a netsave
+ */
+void DoAutoOrNetsave(int& counter, bool netsave) {
```
Comment (glx22) [inline src/saveload/saveload.cpp]: Coding style
```suggestion
void DoAutoOrNetsave(int &counter, bool netsave) {
```

File: src/saveload/saveload.h [hunk 6]
```diff
@@ -377,6 +377,8 @@ void WaitTillSaved();
 void ProcessAsyncSaveFinish();
 void DoExitSave();
 
+void DoAutoOrNetsave(int& counter, bool netsave = false);
```
Comment (glx22) [inline src/saveload/saveload.h]: Same here.

**Comment by Taschi120** [comment]
I'm have no idea why the commit checker is failing, the commit message format looks fine to me?

**Comment by LordAro** [comment]
There's a space at the beginning of the line

File: src/saveload/saveload.cpp [hunk 7]
```diff
@@ -2982,6 +2982,39 @@ SaveOrLoadResult SaveOrLoad(const std::string &filename, SaveLoadOperation fop,
 	}
 }
 
+/**
+ * Create an autosave or netsave.
+ * @param counter A reference to the counter variable to be used for rotating the file name
+ * @param netsave indicates if this is a regular autosave or a netsave
+ */
+void DoAutoOrNetsave(int &counter, bool netsave) {
```
Comment (LordAro) [inline src/saveload/saveload.cpp]: Opening `{` goes on a newline

**Review by glx22 — DISMISSED** [review]
I'll fix the `{` in #9397

**Review by TrueBrain — APPROVED** [review]
Sorry @glx22 ; I also fixed some dots, capitals, and more of that nitpicking stuff. Squashed the commit and altered the commit message to make changelog easier :)

**Comment by TrueBrain** [comment]
Tnx for this @Taschi120 , a fine contribution to OpenTTD :D

**Comment by Taschi120** [comment]
I'll pop into the chat tomorrow and would be grateful if you could let me know in detail what exactly the "nitpicky" things are so I can do better in future. :)

**Pull request merged**